demos/go/gin/main.go

```diff
@@ -0,0 +1,166 @@
+
+	if err != nil {
+		log.Fatal("Error loading env files")
+	}
+	// Set the router as the default one shipped with Gin
+	router := gin.Default()
+
+	// Serve frontend static files
+	router.Use(static.Serve("/", static.LocalFile("../../../theme", true)))
+
+	router.GET("/api", func(c *gin.Context) { //api routing
+		c.JSON(http.StatusOK, gin.H{
+			"message": "Successfully started",
+		})
+	})
+	router.GET("/api/profile", handleget)   //Get request
+	router.POST("/api/profile", handlepost) //Post Request
+	// Start and run the server
+	router.Run(":3000")
+
+}
+func handleget(c *gin.Context) {
+	var w http.ResponseWriter = c.Writer
+	var errors string
+	var out Output
+	out.Message = "an error occoured"
+	out.Status = "error"
+
+	token := c.Request.URL.Query().Get("token")
+	respCode := 200
+
+	cfg := lr.Config{
+		ApiKey:    os.Getenv("APIKEY"),
+		ApiSecret: os.Getenv("APISECRET"),
+	}
+	lrclient, err := lr.NewLoginradius(
+		&cfg,
+		map[string]string{"token": token},
+	)
+	if err != nil {
+		errors = errors + err.(lrerror.Error).OrigErr().Error()
+		respCode = 500
+	}
+
+	res, err := lrauthentication.Loginradius(lrauthentication.Loginradius{lrclient}).GetAuthReadProfilesByToken()
+	if err != nil {
+		errors = errors + err.(lrerror.Error).OrigErr().Error()
+		respCode = 500
+	}
+
+	var data map[string]interface{}
+	er := json.Unmarshal([]byte(res.Body), &data)
+	if er != nil {
+		//panic(err)
+	}
+	if data["Uid"] != nil {
+		out.Data = data
+		out.Message = "Profile fetched"
+		out.Status = "success"
+	} else {
+		out.Message = "Account does not exist."
+	}
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(respCode)
+	if errors != "" {
+		log.Printf(errors)
+		w.Write([]byte(errors))
+	}
+	b, e := json.Marshal(out)
+	if e != nil {
+		//panic(err)
+	}
+	w.Write([]byte(b))
+}
+
+func handlepost(c *gin.Context) {
+	var w http.ResponseWriter = c.Writer
+	var errors string
+	respCode := 200
+	var out Output
+	out.Message = "an error occoured"
+	out.Status = "error"
+
+	cfg := lr.Config{
+		ApiKey:    os.Getenv("APIKEY"),
+		ApiSecret: os.Getenv("APISECRET"),
+	}
+	token := c.Request.URL.Query().Get("token")
+
+	lrclient, err := lr.NewLoginradius(
+		&cfg,
+		map[string]string{"token": token},
+	)
+	if err != nil {
+		errors = errors + err.(lrerror.Error).OrigErr().Error()
+		respCode = 500
+	}
+
+	data := struct {
+		FirstName string
+		LastName  string
+		About     string
+	}{}
+
+	b, _ := ioutil.ReadAll(c.Request.Body)
+	json.Unmarshal(b, &data)
+
+	response, err := lrauthentication.Loginradius(lrauthentication.Loginradius{lrclient}).PutAuthUpdateProfileByToken(
+		data,
+	)
+
+	if err != nil {
+		errors = errors + err.(lrerror.Error).OrigErr().Error()
+		respCode = 500
+	} else {
+		var data map[string]interface{}
+		er := json.Unmarshal([]byte(response.Body), &data)
+		if er != nil {
+			//panic(err)
+		}
+		out.Message = "Profile has been updated successfully."
+		out.Status = "success"
+		out.Data = data
```

Comment:
@gagan3012 Great, now looks good.
Only a minor issue I've found is that profiles are coming in Data Property.
i.e response will be like of this `PutAuthUpdateProfileByToken` function
```
"Body" : {
  "IsPosted":true,
   "Data":{ 
    .....
   }
}
```
So you can check if `IsPosted` property is true and send the profile response in `out.Data`

Get the reference from node js -> express demo also let me know if you need any help in this.

Reply:
@pareek-narendra  I have added the check for IsPosted. Thank you!

Reply:
@gagan3012 
That is fine and look good but read my last comment properly. I have mentioned one more issue.
`Only a minor issue I've found is that profiles are coming in Data Property.`
Right now you are unmarshalling `response.Body` in line no. 145 but the profile is coming in `response.Body.Data`.
So you need to send the profile in `out.Data`  as the direct profile is handled on the frontend.

Reply:
@pareek-narendra 
Thank you I have fixed it now! Please take a look!